### ISSUE TITLE
Unable to disconnect after reconnecting

### ISSUE THREAD

**Issue by oscarhermoso** [tncrazvan/sveltekit-sse#43]
## Reproduce

Code that reproduces the issue looks something like this:

```ts
  // +page.ts
  export function load() {
    let reconnect
    const connection = source(`/room/waiting-lobby`, {
      close({ connect }) { connect = reconnect } })
    })
    const events = connection.select('events').json()
    return { events, connection, reconnect }
  }
```
```svelte
<!-- +page.svelte -->
<script>
  export let data
</script>

{data.events}

<button on:click={data.connection.close}>Disconnect</button>
<button on:click={data.reconnect}>Reconnect</button>
```

1. Load page
2. Click disconnect
3. Click reconnect
4. Click disconnect

Full code repro also raised as this PR, https://github.com/tncrazvan/sveltekit-sse/pull/44

## Expected

Page loads, disconnects, reconnects, and then disconnects again

## Actual

Page loads, disconnects, reconnects, and then is no longer able to disconnect by calling `data.connection.close()`

**Comment by tncrazvan** [comment]
Hello @oscarhermoso ,

This should now be fixed, please update to version [0.12.9](https://github.com/tncrazvan/sveltekit-sse/releases/tag/0.12.9) to get the changes.

I've tried your reproduction and it seems to work as intended (can you confirm it's the intended behavior?)

![Peek 2024-05-11 01-42](https://github.com/tncrazvan/sveltekit-sse/assets/6891346/ca54937b-a7c4-4556-b1f4-df229502f4f8)

Let me know if this addresses your issue.

**Comment by oscarhermoso** [comment]
Works great! Thanks again @tncrazvan
